packages/cli/src/scaling/pubsub/publisher.service.ts

```diff
@@ -46,6 +46,9 @@ export class Publisher {
 
 	/** Publish a command into the `n8n.commands` channel. */
 	async publishCommand(msg: Omit<PubSub.Command, 'senderId'>) {
+		// @TODO: Once this class is only ever used in scaling mode, remove next line.
+		if (config.getEnv('executions.mode') !== 'queue') return;
```

Comment:
it feels like we are repeating this check. Since the execution mode can't change at runtime, could we not change `await this.client.publish` to `await this.client?.publish` instead ?

Reply:
Not a fan of either execution mode sanity checks or calling methods with an uninitialized client. This is only a temporary fix until we can decouple and dynamically load the `Publisher` as mentioned [here](https://n8nio.slack.com/archives/C069HS026UF/p1729580093275339?thread_ts=1729526419.136009&cid=C069HS026UF).